File: scratch/search_colors.py
import os
import re

pattern = re.compile(r'color:\s*"(white|rgba\(255,\s*255,\s*255,\s*[0-9.]+\))"')
bg_pattern = re.compile(r'background:\s*"(rgba\(13,\s*6,\s*40,\s*[0-9.]+\))"')

matches = []
for root, dirs, files in os.walk('src'):
    for f in files:
        if f.endswith('.tsx'):
            path = os.path.join(root, f)
            with open(path, 'r', encoding='utf-8') as file:
                for i, line in enumerate(file, 1):
                    if pattern.search(line) or bg_pattern.search(line):
                        matches.append(f"{path}:{i} -> {line.strip()}")

print(f"Total matches: {len(matches)}")
for m in matches[:50]:
    print(m)

File: scratch/search_colors.js
const fs = require('fs');
const path = require('path');

const colorRegex = /color:\s*"(white|rgba\(255,\s*255,\s*255,\s*[0-9.]+\))"/g;
const bgRegex = /background:\s*"(rgba\(13,\s*6,\s*40,\s*[0-9.]+\))"/g;

const matches = [];

function walk(dir) {
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat && stat.isDirectory()) {
      if (file !== 'node_modules' && file !== '.next') {
        walk(fullPath);
      }
    } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
      const content = fs.readFileSync(fullPath, 'utf8');
      const lines = content.split('\n');
      lines.forEach((line, idx) => {
        colorRegex.lastIndex = 0;
        bgRegex.lastIndex = 0;
        if (colorRegex.test(line) || bgRegex.test(line)) {
          matches.push(`${fullPath}:${idx + 1} -> ${line.trim()}`);
        }
      });
    }
  });
}

walk('src');
console.log(`Total matches: ${matches.length}`);
matches.slice(0, 100).forEach(m => console.log(m));
